transcom/mymove
# B 20621 mobile shipment apis int 2

## [B-20620](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A982508)

## Summary

This is to add/modify services and Internal api endpoints related to Mobile Home Shipment on customer's side of the app.

- Added Mobile Home Shipment services(updater, fetcher, creator)
- Updated MTOShipment services to include mobile home shipment.
- Added mobile home shipment to MTOShipment internal API(GET, POST, PATCH).
- Feature Flag added to Internal, GHC, and Prime MTOShipment  API Handlers(GET).

### How to test
1. Create a move with any shipment.
2. As a customer ->  click "Go to Move" for the current move.
3. Use the browser dev tools get the info you need for the Swagger or Postman.

Use Swagger or Postman to check Internal APIs:

Import `internal.yaml` file from swagger folder if you have not already.

POST - http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/createMTOShipment
Example body to use:
```
{
  "moveTaskOrderID": "a615abd5-c64a-4c0d-b438-2b5e66790130",
  "shipmentType": "MOBILE_HOME",
  "mobileHomeShipment": {
    "year": 1111,
    "model": "Mobile Home Model",
    "make": "Mobile Home Make",
    "lengthInInches": 169,
    "widthInInches": 83,
    "heightInInches": 78,
  }
}
```

PATCH - http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/updateMTOShipment
Example body to use:
```
{
  "moveTaskOrderID": "2ceb3116-b24b-41df-a023-67ac09ec13de",
  "shipmentType": "MOBILE_HOME",
  "mobileHomeShipment": {
    "model": "Mobile Home Model 2"
  }
}
```

GET - http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/listMTOShipments

GET - http://milmovelocal:3000/swagger-ui/internal.html#/moves/getAllMoves


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
_Grab moveTaskOrder ID from developer tools_
<img width="531" alt="Screenshot 2024-08-13 at 11 19 04 AM" src="https://github.com/user-attachments/assets/860d2c3a-6ea0-4445-bba9-e0debfc153da">

_Example of swagger input for mobile home POST_
<img width="1453" alt="Screenshot 2024-08-13 at 11 18 13 AM" src="https://github.com/user-attachments/assets/c15e14e5-e573-434f-a7d6-104bbbbcea81">

_Mobile Home model returned after POST_
<img width="1440" alt="Screenshot 2024-08-13 at 11 18 05 AM" src="https://github.com/user-attachments/assets/fc24ce44-0814-40f1-b809-24bb20a5ebdc">

_Mobile Home model returned after PATCH_
<img width="1192" alt="Screenshot 2024-08-13 at 11 59 41 AM" src="https://github.com/user-attachments/assets/6c73d537-97f1-42e5-a1ed-931e3a108c44">
